### DISCUSSION TITLE
Splitting a dimension into two

### DISCUSSION THREAD

**Discussion by lemartinet** [pydata/xarray#6527] [Q&A] [answered]
Hi all!

I'm still learning to manipulate xarrays so forgive me if this is a trivial question!
Let's say I have an xarray with one dimension "x:y" with coordinates ["x1:y1", "x1:y2", "x1:y4", ...] representing a grid but flattened.
How do I go about to turning this into a two dimension array, with dim "x" ["x1", "x2", ...] and dim "y" ["y1", "y2", ...]
Does that make sense?

Thanks!

**Comment by dcherian** [comment, marked as answer]
You could 
1. use [coarsen.construct](https://docs.xarray.dev/en/stable/generated/xarray.core.rolling.DataArrayCoarsen.construct.html)
2. Try the MultiIndex + unstack approach: https://github.com/pydata/xarray/discussions/5119

**Reply by lemartinet** [reply]
Thanks a lot for the tip #2, that did the job! I was trying to do smtg around that idea but were misusing `assign_coords`
